src/internal/connector/graph/service.go

```diff
@@ -175,38 +176,30 @@ type clientConfig struct {
 type Option func(*clientConfig)
 
 // populate constructs a clientConfig according to the provided options.
-func (c *clientConfig) populate(opts ...Option) *clientConfig {
-	for _, opt := range opts {
-		opt(c)
+func populateConfig(opts ...Option) *clientConfig {
+	cc := clientConfig{
+		maxRetries: defaultMaxRetries,
+		minDelay:   defaultDelay,
 	}
 
-	return c
-}
-
-// apply updates the http.Client with the expected options.
-func (c *clientConfig) applyMiddlewareConfig() (retry int, delay time.Duration) {
-	retry = defaultMaxRetries
-	if c.overrideRetryCount {
-		retry = c.maxRetries
-	}
-
-	delay = defaultDelay
-	if c.minDelay > 0 {
-		delay = c.minDelay
+	for _, opt := range opts {
+		opt(&cc)
 	}
 
-	return
+	return &cc
 }
 
 // apply updates the http.Client with the expected options.
-func (c *clientConfig) apply(hc *http.Client) {
+func (c *clientConfig) apply(hc *http.Client) *http.Client {
 	if c.noTimeout {
 		// FIXME: This should ideally be 0, but if we set to 0, graph
 		// client with automatically set the context timeout to 0 as
 		// well which will make the client unusable.
 		// https://github.com/microsoft/kiota-http-go/pull/71
 		hc.Timeout = 48 * time.Hour
 	}
+
+	return hc
```

Comment:
Don't think we need a return here. I don't have any strong opinions on it though.